dask/optimize.py

```diff
@@ -517,6 +527,10 @@ def fuse(dsk, keys=None, dependencies=None, ave_width=None, max_width=None,
         _globals.get('fuse_max_width') or
         1.5 + ave_width * math.log(ave_width + 1)
     )
+
+    if not ave_width or not max_height:
+        return dsk, dependencies
```

Comment:
This could be moved up to below the `max_height` normalization if you wanted a slightly faster exit.